twill-yarn/src/main/java/org/apache/twill/internal/appmaster/ApplicationMasterService.java

```diff
@@ -193,7 +200,91 @@ private EventHandler createEventHandler(TwillSpecification twillSpec) throws Cla
     Preconditions.checkArgument(EventHandler.class.isAssignableFrom(handlerClass),
                                 "Class {} does not implements {}",
                                 handlerClass, EventHandler.class.getName());
-    return Instances.newInstance((Class<? extends EventHandler>) handlerClass);
+    final EventHandler delegate = Instances.newInstance((Class<? extends EventHandler>) handlerClass);
+    if (delegate == null) {
+      // if no handler is specified, return an EventHandler with no-op
+      return new EventHandler() {};
+    }
+    // wrap the delegate EventHandler so that all errors will be caught
+    return new EventHandler() {
```

Comment:
You'll need to wrap the `initialize` as well. Also, I'd suggest if `initialize` throw exception, you'll just ignore all calls to other lifecycle methods to avoid excessive logging.